src/bridge/cockpitfsread.c

```diff
@@ -238,6 +249,19 @@ cockpit_fsread_prepare (CockpitChannel *channel)
       return;
     }
 
+  if (fstat (self->fd, &statbuf) < 0)
+    {
+      cockpit_channel_fail (channel, "internal-error",
+                            "%s: couldn't stat: %s", self->path, strerror (errno));
+      return;
+    }
+
+  if (statbuf.st_size > max_read_size)
+    {
```

Comment:
`not-supported` means "Cockpit is not compatible with the software on the system.", I think.  It would be confusing if we ever showed this message.